src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -607,30 +608,33 @@ public bool TryFormat(Span<char> destination, out int charsWritten, [StringSynta
         /// <returns><paramref name="value" /> converted to its nearest representable half-precision floating-point value.</returns>
         public static explicit operator Half(float value)
         {
-            const int SingleMaxExponent = 0xFF;
-
-            uint floatInt = BitConverter.SingleToUInt32Bits(value);
-            bool sign = (floatInt & float.SignMask) >> float.SignShift != 0;
-            int exp = (int)(floatInt & float.BiasedExponentMask) >> float.BiasedExponentShift;
-            uint sig = floatInt & float.TrailingSignificandMask;
-
-            if (exp == SingleMaxExponent)
-            {
-                if (sig != 0) // NaN
-                {
-                    return CreateHalfNaN(sign, (ulong)sig << 41); // Shift the significand bits to the left end
-                }
-                return sign ? NegativeInfinity : PositiveInfinity;
-            }
-
-            uint sigHalf = sig >> 9 | ((sig & 0x1FFU) != 0 ? 1U : 0U); // RightShiftJam
-
-            if ((exp | (int)sigHalf) == 0)
-            {
-                return new Half(sign, 0, 0);
-            }
-
-            return new Half(RoundPackToHalf(sign, (short)(exp - 0x71), (ushort)(sigHalf | 0x4000)));
+            Vector128<uint> v0 = Vector128.CreateScalarUnsafe(0x3880_0000u); //Minimum exponent for rounding
+            Vector128<uint> v1 = Vector128.CreateScalarUnsafe(0x3800_0000u); //Exponent displacement #1
+            Vector128<uint> v3 = Vector128.CreateScalarUnsafe(0x7f80_0000u); //Exponent mask
+            Vector128<uint> v4 = Vector128.CreateScalarUnsafe(0x0680_0000u); //Exponent displacement #2
+            Vector128<float> v5 = Vector128.CreateScalarUnsafe(65520.0f);     //Maximum value that is not Infinity in Half
+            uint v = BitConverter.SingleToUInt32Bits(value);
+            Vector128<float> vval = Vector128.CreateScalarUnsafe(value);
+            vval = Vector128.Abs(vval);  //Clear sign bit
+            uint s = v & 0x8000_0000u;       //Extract sign bit
+            vval = Vector128.Min(v5, vval); //Rectify values that are Infinity in Half
+            Vector128<uint> w = Vector128.Equals(vval, vval).AsUInt32();   //Detecting NaN(a != a if a is NaN)
+            Vector128<uint> y = Vector128.Max(v0, vval.AsUInt32()); //Rectify lower exponent
+            y &= v3;        //Extract exponent
+            y += v4;        //Add exponent by 13
+            Vector128<uint> z = y - v1; //Subtract exponent from y by 112
+            z &= w;         //Zero whole z if value is NaN
+            vval += y.AsSingle();                       //Round Single into Half's precision(NaN also gets modified here, just setting the MSB of fraction)
```

Comment:
style nit, please put a space after `//` and before `(`

it would probably be better to put the comment on the line above the statement in each case.

Reply:
> it would probably be better to put the comment on the line above the statement in each case.

Could you clarify if you are suggesting that the modification for all lines with comments should be handled in this manner?
```csharp
// Round Single into Half's precision (NaN also gets modified here, just setting the MSB of fraction)
value += BitConverter.UInt32BitsToSingle(exponentOffset0);
```

Reply:
Yes, that's my suggestion; it's easier to read.